src/compiler/scala/tools/nsc/typechecker/Implicits.scala

```diff
@@ -90,7 +90,9 @@ trait Implicits {
     if (shouldPrint)
       typingStack.printTyping(tree, "typing implicit: %s %s".format(tree, context.undetparamsString))
     val implicitSearchContext = context.makeImplicit(reportAmbiguous)
-    val result = new ImplicitSearch(tree, pt, isView, implicitSearchContext, pos).bestImplicit
+    val search0 = new ImplicitSearch(tree, pt, isView, implicitSearchContext, pos)
+    val search = pluginsImplicitSearch(search0)
+    val result = pluginsImplicitSearchResult(search.bestImplicit)
```

Comment:
I'm not really comfortable letting plugins change the results of an implicit search. A less powerful extension point would just have `pluginsImplicitSearchResult(search: ImplicitSearch): Unit`.

The standard `ImplicitSearch` can itself contain a timer, which is conditionally computed if stats are on. We don't need to leave that part to the plugin.

Reply:
> I'm not really comfortable letting plugins change the results of an implicit search. A less powerful extension point would just have pluginsImplicitSearchResult(search: ImplicitSearch): Unit.
  
I've adapted it to be a simple hook. `AnalyzerPlugins` is itself a powerful tool that refines types from the typechecker, so I wasn't sure that such a powerful extension point would not be accepted since it aligns with the overall idea of the plugin.
  
> The standard ImplicitSearch can itself contain a timer, which is conditionally computed if stats are on. We don't need to leave that part to the plugin.
  
We already have a timer for this, e.g. `implicitNanos`, and I can access it statically in my compiler plugin, so I think this is not necessary.
  
However, I didn't realize that we have another `ImplicitSearch` for views. They should mostly be unused, but just in case I've added a timer for them too.